docusaurus/docs/develop/envoy/introduction.md

```diff
@@ -222,20 +207,105 @@ sequenceDiagram
     Service->>-Client: 12. Return Response
 ```
 
-## 4. Gateway Endpoint Authorization
+## 4. Specifying the Gateway Endpoint ID
+
+The Auth Server may extract the Gateway Endpoint ID from the request in one of two ways:
+
+1. [URL Path](#221-url-path)
+2. [Header](#222-header)
+
+This is determined by the `ENDPOINT_ID_EXTRACTOR` environment variable in the `auth_server/.env` file.
+
+Valid options are:
+
+- `url_path`
+- `header`
+
+If the `ENDPOINT_ID_EXTRACTOR` environment variable is not set, the default `url_path` extractor is used.
+
+:::warning
+Requests are rejected if either of the following are true:
+
+- The `<GATEWAY_ENDPOINT_ID>` is missing
+- ID is not present in the `Go External Authorization Server`'s `Gateway Endpoint Store`
+:::
+
+:::info
+Regardless of which extractor is used, the Gateway Endpoint ID will always be set in the `x-endpoint-id` header if the reuqest is forwarded to the PATH Service.
+:::
+
+### 4.1 URL Path
+
+When using the `url_path` extractor, the Gateway Endpoint ID must be specified in the URL path.
+
+```
+https://<SERVICE_NAME>.<PATH_DOMAIN>/v1/<GATEWAY_ENDPOINT_ID>
+```
+
+For example, if the `SERVICE_NAME` is `eth` and the `GATEWAY_ENDPOINT_ID` is `a1b2c3d4`:
+
+```
+curl http://anvil.localhost:3001/v1/endpoint_3 \
+  -X POST \
+  -H "Content-Type: application/json" \
+  -d '{"jsonrpc": "2.0", "id": 1, "method": "eth_blockNumber" }'
+```
+
+### 4.2 Header
+
+When using the `header` extractor, the Gateway Endpoint ID must be specified in the `x-endpoint-id` header.
+
+```
+-H "x-endpoint-id: <GATEWAY_ENDPOINT_ID>"
+```
+
+For example, if the `x-endpoint-id` header is set to `a1b2c3d4`:
+
+```
+curl http://anvil.localhost:3001/v1 \
+  -X POST \
+  -H "Content-Type: application/json" \
+  -H "x-endpoint-id: endpoint_3" \
```

Comment:
Wdyt of changing `endpoint_3` to `endpoint_no_auth`.

Ditto regarding the other ones.

Having a semantic name across the board will set a foundation for better docs moving forward.